zerver/lib/message.py

```diff
@@ -242,6 +246,29 @@ def visible_edit_history_for_message(
     return visible_edit_history
 
 
+def topic_resolve_toggled(topic: str, prev_topic: str) -> bool:
+    # Topic unresolved
+    if prev_topic.startswith(RESOLVED_TOPIC_PREFIX) and not topic.startswith(RESOLVED_TOPIC_PREFIX):
+        return prev_topic.lstrip(RESOLVED_TOPIC_PREFIX) == topic
+
+    # Topic resolved
+    if topic.startswith(RESOLVED_TOPIC_PREFIX) and not prev_topic.startswith(RESOLVED_TOPIC_PREFIX):
+        if len(prev_topic) <= MAX_TOPIC_NAME_LENGTH - len(RESOLVED_TOPIC_PREFIX):
+            # When the topic was resolved, it was not truncated,
+            # so we remove the resolved prefix and compare.
+            return topic.lstrip(RESOLVED_TOPIC_PREFIX) == prev_topic
```

Comment:
`lstrip` will remove any sequence of characters that come from the given argument (which is treated as just a set of characters):
```
>>> "a aaaa abcd".lstrip("a ")
'bcd'
```

If this move was resolving the topic, it'll be more specific than that, so this check can be more specific to match:
```suggestion
            return topic[len(RESOLVED_TOPIC_PREFIX) :] == prev_topic
```

Similarly below.

Reply:
Makes sense. Updated to have variables for the length of both constants to use in the string comparisons.